### ISSUE TITLE
Clearer error messages in BatchNormalization

### ISSUE THREAD

**Issue by cmarschner** [Microsoft/CNTK#1687]
GPU version of batch normalization features debug assertions for parameter checking. This needs to be replaced with proper error handling. https://github.com/Microsoft/CNTK/blob/8af30f63f2b033e798bb1b8b5c95cca340646a67/Source/Math/GPUMatrix.cu#L3360
CPU version throws an exception that is not informative (and has a typo). What is the matrix shape, and what is the argument?
https://github.com/Microsoft/CNTK/blob/8af30f63f2b033e798bb1b8b5c95cca340646a67/Source/Math/CPUMatrix.cpp#L4689
